compiler/rustc_span/src/source_map.rs

```diff
@@ -133,7 +133,29 @@ impl FileLoader for RealFileLoader {
         file.read_buf_exact(buf.unfilled())?;
```

Comment:
this is wrong because some files lie and have metadata size much larger than the actual file's size:
e.g. on linux `/sys/class/tty/console/dev` is like that:
full contents:
```
5:1
```
`stat` gives:
```
  File: /sys/class/tty/console/dev
  Size: 4096            Blocks: 0          IO Block: 4096   regular file
Device: 17h/23d Inode: 15051       Links: 1
Access: (0444/-r--r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2023-09-04 17:01:42.101801391 -0700
Modify: 2023-09-04 17:01:42.101801391 -0700
Change: 2023-09-04 17:01:42.101801391 -0700
 Birth: -
```

Reply:
```
error: couldn't read /sys/class/tty/console/dev: failed to fill buffer
 --> demo.rs:1:22
  |
1 | const BYTES: &[u8] = include_bytes!("/sys/class/tty/console/dev");
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `include_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error
```
I've been wondering if there was such a case! I'll adjust this in a bit...
(does this work here?)
@rustbot author